src/compiler/checker.ts

```diff
@@ -38460,19 +38459,26 @@ namespace ts {
 
         function checkTestingKnownTruthyCallableOrAwaitableType(condExpr: Expression, condType: Type, body?: Statement | Expression) {
             if (!strictNullChecks) return;
+            bothHelper(condExpr, body);
 
-            helper(condExpr, body);
-            while (isBinaryExpression(condExpr) && condExpr.operatorToken.kind === SyntaxKind.BarBarToken) {
-                condExpr = condExpr.left;
+            function bothHelper(condExpr: Expression, body: Expression | Statement | undefined) {

```

Comment:
todo: this could use a better name, the `helper` itself probably could use a better name too. I can try to figure out something better later on